src/main/java/model/Winning.java

```diff
@@ -0,0 +1,58 @@
+package model;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public enum Winning {
+    FAIL(0, 0, 0, ""),
+    FIFTH(3, 5000, 0, "3개 일치 (5,000원)"),
+    FOURTH(4, 50000, 0, "4개 일치 (50,000원)"),
+    THIRD(5, 1500000, 0, "5개 일치 (1,500,000원)"),
+    SECOND(5, 30000000, 1, "5개 일치, 보너스 볼 일치 (30,000,000원)"),
+    FIRST(6, 2000000000, 0, "6개 일치 (2,000,000,000원)");
+
+    private final int count;
+    private final long prize;
+    private final int hasBonusNumber;
+    private final String description;
+
+    Winning(int count, long prize, int hasBonusNumber, String description) {
+        this.count = count;
+        this.prize = prize;
+        this.hasBonusNumber = hasBonusNumber;
+        this.description = description;
+    }
+
+    public static Winning matchWinning(int count, int hasBonusNumber) {
+        Winning result = Arrays.stream(Winning.values())
+                .filter(w -> w.count == count)
+                .findFirst()
+                .orElse(Winning.FAIL);
+
+        if (result == THIRD && hasBonusNumber == SECOND.hasBonusNumber) {
+            result = SECOND;
+        }
+
+        return result;
+    }
+
+    public static long calculatePrize(Winning winning, int count) {
+        final Optional<Winning> target = Arrays.stream(Winning.values()).filter(w -> w.equals(winning)).findFirst();
+
+        if (target.isPresent()) {
+            return target.get().prize * count;
+        }
+
+        return 0;
+    }
```

Comment:
Optional 의 isPresent 를 사용하셨네요.
해당 로직은 소나린트에 걸리는데
```suggestion
 return target.map(value -> value.prize * count)
                .orElse(0L);
```

로 변경하는게 더 좋을 듯 합니다.

Reply:
변경 후 더 정리할 수 있을 거 같아서 정리했는데 저번에 리뷰해주신 stream의 가독성 부분이 약간 걸립니다 ㅠㅠ
한 번 코드 확인해주시면 감사하겠습니다!!